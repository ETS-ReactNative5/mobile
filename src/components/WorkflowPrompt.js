import React, { Component } from 'react'
import {
  View
} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import StyledModal from './StyledModal'
import StyledText from './StyledText'
import Button from './Button'
import { addIndex, map } from 'ramda'
import PropTypes from 'prop-types';

export class WorkflowPrompt extends Component {
  render() {
    const renderButtonGroup = (workflows, isSwipe) => {
      return (
        <View style={styles.buttonsContainer}>
          { addIndex(map)(
            (workflow, idx) => {
              return renderButton(workflow, idx, isSwipe)
            },
            workflows
          ) }
        </View>
      )
    }

    const renderButton = (workflow, idx, isMobile) => {
      const chooseWorkflow = (workflow) => {
        this.props.hideWorkflowPrompt()
        if (isMobile) {
          this.props.openMobileProject(workflow.id)
        } else {
          this.props.openExternalProject()
        }
      }
      return (
        <Button
          key={idx}
          text={ workflow.display_name }
          additionalStyles={[styles.button]}
          additionalTextStyles={[styles.buttonText]}
          handlePress={ ()=> { chooseWorkflow(workflow) }}
        />
      )
    }

    return (
      <StyledModal
        isVisible={this.props.isVisible}
        setVisibility={this.props.hideWorkflowPrompt}>
        <StyledText
          additionalStyles={[styles.header]}
          text={ 'Please Choose Workflow' } />

        <StyledText
          textStyle={'headerText'}
          additionalStyles={[styles.prompt]}
          text={ 'FOR MOBILE' } />

        { renderButtonGroup(this.props.mobileWorkflows, true) }

        <StyledText
          textStyle={'headerText'}
          additionalStyles={[styles.prompt]}
          text={ 'OTHER WORKFLOWS' } />

        { renderButtonGroup(this.props.nonMobileWorkflows, false) }

        <Button
          handlePress={ this.props.hideWorkflowPrompt }
          additionalStyles={[styles.cancelButton]}
          buttonStyle={'navyButton'}
          text={'Cancel'} />
      </StyledModal>
    )
  }
}

const styles = EStyleSheet.create({
  header: {
    fontSize: 20,
    fontWeight: '500',
    color: '$darkGrey',
  },
  subheader: {
    color: '$grey',
  },
  prompt: {
    fontWeight: '500',
    fontSize: 14,
    marginTop: 20,
  },
  buttonsContainer: {
    marginBottom: 10,
    marginLeft: 8,
  },
  button: {
    backgroundColor: 'white',
    borderWidth: 1,
    borderColor: '$darkGreyTextColor'
  },
  buttonText: {
    color: '$darkGreyTextColor'
  },
  link: {
    backgroundColor: 'white',
    borderWidth: 0,
    padding: 5,
    justifyContent: 'flex-start',

  },
  cancelButton: {
    marginTop: 20
  },
});

WorkflowPrompt.propTypes = {
  mobileWorkflows: PropTypes.array,
  nonMobileWorkflows: PropTypes.array,
  isVisible: PropTypes.bool,
  openMobileProject: PropTypes.func.isRequired,
  openExternalProject: PropTypes.func.isRequired,
  hideWorkflowPrompt: PropTypes.func.isRequired,
}

export default WorkflowPrompt
