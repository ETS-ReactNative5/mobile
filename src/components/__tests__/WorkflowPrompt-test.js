import 'react-native'
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow';
import WorkflowPrompt from '../WorkflowPrompt'

const mobileWorkflows=[
  {id: 1, display_name: 'Mobile 1'},
  {id: 2, display_name: 'Mobile 2'}
]

const nonMobileWorkflows=[
  {id: 1, display_name: 'Non 1'},
  {id: 2, display_name: 'Non 2'}
]

const renderer = new ShallowRenderer();
it('renders correctly when visible', () => {
  const tree = renderer.render(
      <WorkflowPrompt
        mobileWorkflows={mobileWorkflows}
        nonMobileWorkflows={nonMobileWorkflows}
        isVisible={true}
        openMobileProject={jest.fn}
        openExternalProject={jest.fn}
        hideWorkflowPrompt={jest.fn}
      />
  )
  expect(tree).toMatchSnapshot()
})

it('passes false to Modal when not visible', () => {
  const tree = renderer.render(
    <WorkflowPrompt
      mobileWorkflows={mobileWorkflows}
      nonMobileWorkflows={nonMobileWorkflows}
      isVisible={false}
      openMobileProject={jest.fn}
      openExternalProject={jest.fn}
      hideWorkflowPrompt={jest.fn}
    />
  )
  expect(tree).toMatchSnapshot()
})
