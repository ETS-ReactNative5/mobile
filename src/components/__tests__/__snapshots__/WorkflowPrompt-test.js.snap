exports[`test passes false to Modal when not visible 1`] = `
<View>
  <Modal
    animationType="fade"
    onRequestClose={[Function]}
    transparent={true}
    visible={false}>
    <View
      style={undefined}>
      <ScrollView
        contentContainerStyle={undefined}
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              Array [
                undefined,
              ],
            ]
          }>
          Please Choose Workflow
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
              Array [
                undefined,
              ],
            ]
          }>
          FOR MOBILE
        </Text>
        <View
          style={undefined}>
          <TouchableOpacity
            activeOpacity={0.5}
            disabled={false}
            onPress={[Function]}
            style={
              Array [
                undefined,
                Array [
                  undefined,
                ],
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                  ],
                ]
              }>
              Mobile 1
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.5}
            disabled={false}
            onPress={[Function]}
            style={
              Array [
                undefined,
                Array [
                  undefined,
                ],
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                  ],
                ]
              }>
              Mobile 2
            </Text>
          </TouchableOpacity>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
              Array [
                undefined,
              ],
            ]
          }>
          OTHER WORKFLOWS
        </Text>
        <View
          style={undefined}>
          <TouchableOpacity
            activeOpacity={0.5}
            disabled={false}
            onPress={[Function]}
            style={
              Array [
                undefined,
                Array [
                  undefined,
                ],
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                  ],
                ]
              }>
              Non 1
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.5}
            disabled={false}
            onPress={[Function]}
            style={
              Array [
                undefined,
                Array [
                  undefined,
                ],
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                  ],
                ]
              }>
              Non 2
            </Text>
          </TouchableOpacity>
        </View>
        <TouchableOpacity
          activeOpacity={0.5}
          disabled={false}
          onPress={[Function]}
          style={
            Array [
              Array [
                undefined,
                undefined,
              ],
              Array [
                undefined,
              ],
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}>
            Cancel
          </Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      style={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        ÔÄç
      </Text>
    </TouchableOpacity>
  </Modal>
</View>
`;

exports[`test renders correctly when visible 1`] = `
<View>
  <Modal
    animationType="fade"
    onRequestClose={[Function]}
    transparent={true}
    visible={true}>
    <View
      style={undefined}>
      <ScrollView
        contentContainerStyle={undefined}
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              Array [
                undefined,
              ],
            ]
          }>
          Please Choose Workflow
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
              Array [
                undefined,
              ],
            ]
          }>
          FOR MOBILE
        </Text>
        <View
          style={undefined}>
          <TouchableOpacity
            activeOpacity={0.5}
            disabled={false}
            onPress={[Function]}
            style={
              Array [
                undefined,
                Array [
                  undefined,
                ],
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                  ],
                ]
              }>
              Mobile 1
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.5}
            disabled={false}
            onPress={[Function]}
            style={
              Array [
                undefined,
                Array [
                  undefined,
                ],
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                  ],
                ]
              }>
              Mobile 2
            </Text>
          </TouchableOpacity>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
              undefined,
              Array [
                undefined,
              ],
            ]
          }>
          OTHER WORKFLOWS
        </Text>
        <View
          style={undefined}>
          <TouchableOpacity
            activeOpacity={0.5}
            disabled={false}
            onPress={[Function]}
            style={
              Array [
                undefined,
                Array [
                  undefined,
                ],
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                  ],
                ]
              }>
              Non 1
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.5}
            disabled={false}
            onPress={[Function]}
            style={
              Array [
                undefined,
                Array [
                  undefined,
                ],
              ]
            }>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  undefined,
                  Array [
                    undefined,
                  ],
                ]
              }>
              Non 2
            </Text>
          </TouchableOpacity>
        </View>
        <TouchableOpacity
          activeOpacity={0.5}
          disabled={false}
          onPress={[Function]}
          style={
            Array [
              Array [
                undefined,
                undefined,
              ],
              Array [
                undefined,
              ],
            ]
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={undefined}>
            Cancel
          </Text>
        </TouchableOpacity>
      </ScrollView>
    </View>
    <TouchableOpacity
      activeOpacity={0.5}
      onPress={[Function]}
      style={undefined}>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontFamily": "FontAwesome",
              "fontSize": 12,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            undefined,
          ]
        }>
        ÔÄç
      </Text>
    </TouchableOpacity>
  </Modal>
</View>
`;
